sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/query.sql

```diff
@@ -224,16 +224,20 @@ first_seen_date AS (
 -- The next CTE returns the second_seen_date calculated as the next date reported by the
 --  main ping after first_seen_date or NULL. Dates reported by other pings are excluded.
 second_seen_date AS (
-   SELECT
+  SELECT
     client_id,
-    all_dates[SAFE_OFFSET(0)] AS second_seen_date
+    MIN(seen_dates) AS second_seen_date
   FROM
     main_ping
+  LEFT JOIN
+    UNNEST(all_dates) as seen_dates
   LEFT JOIN
     first_seen_date fs
     USING (client_id)
   WHERE
-    main_ping.all_dates[SAFE_OFFSET(0)] > fs.first_seen_date
```

Comment:
Both SAFE_OFFSET(0) and SAFE_OFFSET(1) return wrong dates depending on which main ping is reporting the second date.